app/src/main/java/no/usn/mob3000/data/repository/social/FriendsRepository.kt

```diff
@@ -0,0 +1,109 @@
+package no.usn.mob3000.data.repository.social
+
+import no.usn.mob3000.data.model.social.FriendsDto
+import no.usn.mob3000.data.model.social.ProfileDto
+import no.usn.mob3000.data.source.remote.social.FriendRequestDataSource
+import no.usn.mob3000.data.source.remote.social.FriendsDataSource
+import no.usn.mob3000.domain.model.auth.UserProfile
+import no.usn.mob3000.domain.model.social.FriendData
+import no.usn.mob3000.domain.repository.social.IFriendsRepository
+
+/**
+ *
+ * This repository orchestrates mainly the showcasing of the users friends. In difference to [FriendRequestRepository], this repository
+ * contains only fetch-actions.
+ *
+ * Mapping is implemented to join [displayName] from [UserProfile] to [FriendData].
+ * TODO: Deletion of a friendship and other actions if we have the time
+ *
+ * @param FriendsDataSource The data source for fetching friends.
+ * @param profileUserDataSource The data source for fetching non-friend profiles.
+ * @author Husseinabdulameer11, 258030
+ * @created 2024-11-05
+ **/
+class FriendsRepository (
+    private val friendsDataSource: FriendsDataSource = FriendsDataSource(),
+    private val friendRequestDataSource: FriendRequestDataSource = FriendRequestDataSource(),
+): IFriendsRepository{

```

Comment:
`FriendRepository` does not expect `IFriendsRepository` as a result. It's implementing an interface and needs to be represented as such following Kotlin's style guide. It should be `) : IF`. Additionally, the opening bracket for the class must have a space between the interface and itself. It should be `ory {`.